mobile-app/app/screens/AppNavigator/screens/Balances/hooks/TokenPrice.ts

```diff
@@ -0,0 +1,120 @@
+import { useEffect } from 'react'
+import BigNumber from 'bignumber.js'
+import { RootState } from '@store'
+import { fetchPoolPairs } from '@store/wallet'
+import { PoolPairData } from '@defichain/whale-api-client/dist/api/poolpairs'
+import { ActivePrice } from '@defichain/whale-api-client/dist/api/prices'
+import { useWhaleApiClient } from '@shared-contexts/WhaleContext'
+import { useDispatch, useSelector } from 'react-redux'
+import { getActivePrice } from '@screens/AppNavigator/screens/Auctions/helpers/ActivePrice'
+import { checkIfPair, findPath, GraphProps } from '@screens/AppNavigator/screens/Dex/helpers/path-finding'
+
+interface DexTokenPrice {
+  getTokenPrice: (symbol: string, activePrice?: ActivePrice) => BigNumber
+  calculatePriceRates: (fromTokenSymbol: string, pairs: PoolPairData[], amount: string) => { aToBPrice: BigNumber, bToAPrice: BigNumber, estimated: BigNumber }
+  getSelectedPoolPairs: (tokenASymbol: string, tokenBSymbol: string) => PoolPairData[]
+  getPairAmountFromLP: (symbol: string, amount: string) => { tokenAAmount: BigNumber, tokenBAmount: BigNumber, tokenASymbol: string, tokenBSymbol: string }
+}
+
+export function useTokenPrice (): DexTokenPrice {
+  const client = useWhaleApiClient()
+  const dispatch = useDispatch()
+  const blockCount = useSelector((state: RootState) => state.block.count)
+  const pairs = useSelector((state: RootState) => state.wallet.poolpairs)
+
+  const graph: GraphProps[] = pairs.map(pair => {
+    return {
+      pairId: pair.data.id,
+      a: pair.data.tokenA.symbol,
+      b: pair.data.tokenB.symbol
+    }
+  })
+
+  useEffect(() => {
+    dispatch(fetchPoolPairs({ client }))
+  }, [blockCount])
+
+  function getPairAmountFromLP (symbol: string, amount: string): { tokenAAmount: BigNumber, tokenBAmount: BigNumber, tokenASymbol: string, tokenBSymbol: string } {
+      const pair = pairs.find(pair => pair.data.symbol === symbol)
+
+      if (pair === undefined) {
+        throw new Error(`The LP symbol is not existing: ${symbol}`)
+      }
+
+      const ratioToTotal = new BigNumber(amount).div(pair.data.totalLiquidity.token)
+      // assume defid will trim the dust values too
+      return {
+        tokenAAmount: ratioToTotal.times(pair.data.tokenA.reserve).decimalPlaces(8, BigNumber.ROUND_DOWN),
+        tokenBAmount: ratioToTotal.times(pair.data.tokenB.reserve).decimalPlaces(8, BigNumber.ROUND_DOWN),
+        tokenASymbol: pair.data.tokenA.symbol,
+        tokenBSymbol: pair.data.tokenB.symbol
+      }
+  }
+
+  function getTokenPrice (symbol: string, activePrice?: ActivePrice): BigNumber {
+    // active price for collateralTokens
+    if (activePrice !== undefined) {
```

Comment:
This is not a good indicator of collateralTokens. Can we have a flag or something?